great_expectations/analytics/base_event.py

```diff
@@ -36,17 +36,21 @@ def data_context_id(self) -> UUID:
         return get_config().data_context_id
 
     @property
-    def organization_id(self) -> UUID:
+    def organization_id(self) -> Optional[UUID]:
         return get_config().organization_id
 
     @property
     def oss_id(self) -> UUID:
         return get_config().oss_id
 
     @property
-    def user_id(self) -> UUID:
+    def user_id(self) -> Optional[UUID]:
         return get_config().user_id
 
+    @property
+    def distinct_id(self) -> UUID:
+        return self.user_id or self.oss_id
```

Comment:
Could we add a docstring here just to clarify what this entails? Why do we prioritize `user_id` over `oss_id`?

Reply:
I added a comment that I hope addresses this fully, but happy to revise again. Basically, the user_id is more likely to be correct than a generic (non user specific) OSS ID